packages/smooth_app/lib/pages/multi_select_product_page.dart

```diff
@@ -0,0 +1,245 @@
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/data_models/user_preferences.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+
+/// Page where products can be selected for copy or removal
+///
+/// The list of products come from
+/// a pantry, a shopping list or a product list.
+class MultiSelectProductPage extends StatefulWidget {
+  const MultiSelectProductPage.pantry({
+    @required this.barcode,
+    @required this.pantries,
+    this.index,
+    this.pantryType,
+  }) : productList = null;
+
+  const MultiSelectProductPage.productList({
+    @required this.barcode,
+    this.productList,
+  })  : pantries = null,
+        pantryType = null,
+        index = null;
+
+  /// Initial selected barcode
+  final String barcode;
+
+  final List<Pantry> pantries;
+  final int index;
+  final PantryType pantryType;
+  Pantry get pantry => pantries == null ? null : pantries[index];
+
+  final ProductList productList;
+
+  @override
+  _MultiSelectProductPageState createState() => _MultiSelectProductPageState();
+}
+
+class _MultiSelectProductPageState extends State<MultiSelectProductPage> {
+  final Set<String> _selectedBarcodes = <String>{};
+  List<String> _orderedBarcodes; // late final
+
+  @override
+  void initState() {
+    super.initState();
+    _selectedBarcodes.add(widget.barcode);
+    if (widget.productList != null) {
+      _orderedBarcodes = widget.productList.getOrderedBarcodes();
+    } else {
+      _orderedBarcodes = widget.pantry.getOrderedBarcodes();
+    }
+  }
+
+  void _removeBarcode(final String barcode) => widget.productList != null
+      ? widget.productList.barcodes.remove(barcode)
+      : widget.pantry.removeBarcode(barcode);
+
+  Product _getProduct(final String barcode) => widget.productList != null
+      ? widget.productList.getProduct(barcode)
+      : widget.pantry.products[barcode];
+
+  Future<void> _commit(
+    final UserPreferences userPreferences,
+    final DaoProductList daoProductList,
+  ) async =>
+      widget.productList != null
+          ? await daoProductList.put(widget.productList)
+          : await Pantry.putAll(
+              userPreferences,
+              widget.pantries,
+              widget.pantryType,
+            );
+
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
+    final UserPreferences userPreferences = context.watch<UserPreferences>();
+    final DaoProductList daoProductList = DaoProductList(localDatabase);
+    final DaoProduct daoProduct = DaoProduct(localDatabase);
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final ThemeData themeData = Theme.of(context);
+    final Size screenSize = MediaQuery.of(context).size;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('${_selectedBarcodes.length} selected'),
```

Comment:
Please add all the Strings to lib/app_en,arb for the translation

Reply:
If that's OK with you, I'll skip that part.

Whatever I'm developing for the moment is not to be translated immediately in 126 languages, it's just work in progress. And I go back and forth with labels: for instance when I've removed in this PR the bottom navigation bars in product list page and pantry page I guess I removed translated labels. And that's OK with me, because those were temporary widgets.

Of course when the UX gets closer to something interesting, that will be a relevant question.